DEVELOPER.md

```diff
@@ -2,380 +2,515 @@
 
 ```shell
-DEPLOY_WITH=helm make config-transformer deploy
+[k8admin@docd01 vertica-kubernetes]$ docker image ls
+REPOSITORY               TAG            IMAGE ID       CREATED          SIZE
+...
+hadolint/hadolint        2.12.0         12fa10a87864   11 months ago    2.43MB
+...
 ```
 
-To deploy olm all of its prereqs, use the following command:
+## Unit Tests
+
+The unit tests verify both the Helm chart and the VerticaDB operator. Run the unit tests with the following command:
 
 ```shell
-DEPLOY_WITH=olm make setup-olm deploy
+make run-unit-tests
 ```
 
-To remove the operator, regardless of how it was deployed, run the `undeploy` make target:
+This make target pulls the `quintush/helm-unittest` image into your local docker repo:
+
 ```shell
-make undeploy
+[k8admin@docd01 vertica-kubernetes]$ docker image ls
+REPOSITORY               TAG            IMAGE ID       CREATED          SIZE
+...
+quintush/helm-unittest   3.9.3-0.2.11   83c439b2cf46   9 months ago     105MB
+...
 ```
 
-## 6. Running e2e Tests
+Helm chart unit tests are stored in `helm-charts/verticadb-operator/tests` and use the [helm-unittest plugin](https://github.com/quintush/helm-unittest). The helm-unittest repo includes [test samples and templates](https://github.com/quintush/helm-unittest/tree/master/test/data/v3/basic) so you can model your own tests. For details about the test format, see the [helm-unittest GitHub repository](https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md).
 
-The end-to-end (e2e) tests are run through Kubernetes itself. We use kuttl as the testing framework. The operator must be running **as a Kubernetes deployment**, which means the operator container needs to be built and pushed to a repository before starting.
+Unit tests for the VerticaDB operator use the Go testing infrastructure. Some tests run the operator against a mock Kubernetes control plane created with [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest). Per Go standards, test files are stored in package directories and end with `_test.go`.
 
-Ensure that your Kubernetes cluster has a default storageClass. Most of the e2e tests do not specify any storageClass and use the default. For details about setting your storageClass, refer to the [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/).
+## e2e Tests
 
-1. Push the operator with the following command:
+> **IMPORTANT**
+> The e2e tests only run on operators that were [deployed as an object](#deployment-object).
+
+The e2e tests use the [kuttl](https://github.com/kudobuilder/kuttl/) testing framework. Before you begin, set the default StorageClass for your Kubernetes cluster. Otherwise, most of the e2e tests use the `standard (default)` StorageClass. For details, see the [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/).
```

Comment:
The tests will use the default storageClass. The default can change, and in some environments a default isn't even set. If you are running with kind, a suitable default is picked and you don't even need to worry about this.